### DISCUSSION TITLE
Always type hint callbacks in new PRs

### DISCUSSION THREAD

**Discussion by ghost** [laravel/framework#51303] [Ideas] [answered]
Hi,
Laravel is lacking typed parameters and return types in the callbacks it uses. I propose that all new PRs that touch callback should define them because it is way easier to code like that.

One example would be the Query/Builder vs Eloquent/Builder vs Relation sent in where(callback).

Opinions?

**Comment by macropay-solutions** [comment]
Laravel 10 was supposed to be the start of type hinting. But we can still see PRs that introduce NEW code without type hints.
Why is this habbit of not using them in NEW CODE? 

few examples: 
https://github.com/laravel/framework/pull/51805/files
https://github.com/laravel/framework/pull/51773/files
https://github.com/laravel/framework/pull/51582/files

**Comment by ghost** [comment]
Laravel stated that starting from v 10 it ill use typed params and returns.
https://redberry.international/laravel-10-features-and-updates/#mcetoc_1hfrpmpr11v

**Comment by ghost** [comment, marked as answer]
Not according to this https://github.com/laravel/framework/commit/00dfdf48a12a79408d2646ba9540e8636e270e51
